### PR TITLE
Fix origin_session_reuse test

### PR DESCRIPTION
**Pull request by maskit**
The reason of tls_origin_session_reuse failure on autest-master job is TLS version used on the test.

Current test passes only if the connection is TLS 1.3, because a session can be resumed by session id on TLS 1.2.
The box for autest-github has recent OpenSSL and it supports TLS 1.3, so disabling ticket means disabling session reuse.
On the other hand, the box for autest-master has OpenSSL 1.0.2 and it does not support TLS 1.3, so disabling ticket does not mean disabling session reuse.
I confirmed on my laptop that the test fails if I set `proxy.config.ssl.TLSv1_3` to 0 on the test file.

In short, disabling ticket is not enough for disabling session reuse.
On this test, I think we should check that we can disable session reuse at all by setting `origin_session_cache` to 0. In that sense, we may want to have two tests, one for TLS 1.2 and one for TLS 1.3, but I'd like to fix the test failure first.

### CODE REVIEW

**Comment by maskit** [comment]
It's already Friday evening here, and I'll be off next 4 days. Feel free to merge if you approve this.

**Review by duke8253 — APPROVED** [review]
These old OpenSSL versions are annoying.

**Review by bryancall — APPROVED** [review]
